src/crimson/os/seastore/seastore.cc

```diff
@@ -188,17 +189,22 @@ seastar::future<> SeaStore::umount()
 seastar::future<> SeaStore::write_fsid(uuid_d new_osd_fsid)
 {
   LOG_PREFIX(SeaStore::write_fsid);
-  return read_meta("fsid").then([this, FNAME, new_osd_fsid] (auto tuple) {
+  return read_meta("fsid").then([this, FNAME, new_osd_fsid] (auto tuple) mutable {
     auto [ret, fsid] = tuple;
     std::string str_fsid = stringify(new_osd_fsid);
     if (ret == -1) {
+       if (new_osd_fsid.is_zero()) {
+         new_osd_fsid.generate_random();
+       }
+       set_fsid(new_osd_fsid);
```

Comment:
Instead of doing this here, can we just randomize it at the beginning of mkfs?

Reply:
yes, we can, will do it then set_fsid will be called in osd::mkfs.